src/elf/symver.rs

```diff
@@ -59,876 +59,866 @@
+                }
 
-                do_next(self).or_else(|| {
-                    // Adjust current index to count in case of an error.
-                    self.index = self.count;
-                    None
+                Some(Verdaux {
+                    vda_name: vda_name as usize,
+                    vda_next,
                 })
-            }
-        }
+            };
 
-        #[inline]
-        fn size_hint(&self) -> (usize, Option<usize>) {
-            let len = usize::from(self.count - self.index);
-            (0, Some(len))
+            do_next(self).or_else(|| {
+                // Adjust current index to count in case of an error.
+                self.index = self.count;
+                None
+            })
         }
     }
 
-    impl ExactSizeIterator for VerdauxIter<'_> {}
+    #[inline]
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        let len = usize::from(self.count - self.index);
+        (0, Some(len))
+    }
+}
+
+impl ExactSizeIterator for VerdauxIter<'_> {}
 
-    impl FusedIterator for VerdauxIter<'_> {}
+impl FusedIterator for VerdauxIter<'_> {}
 
-    /// An ELF [Version Definition Auxiliary][lsb-verdaux] entry.
-    ///
-    /// [lsb-verdaux]: https://refspecs.linuxbase.org/LSB_5.0.0/LSB-Core-generic/LSB-Core-generic/symversion.html#VERDEFEXTS
-    #[derive(Debug)]
-    pub struct Verdaux {
-        /// Offset to the version or dependency name string in the section header, in bytes.
-        pub vda_name: usize,
-        /// Offset to the next verdaux entry, in bytes.
-        pub vda_next: u32,
-    }
+/// An ELF [Version Definition Auxiliary][lsb-verdaux] entry.
+///
+/// [lsb-verdaux]: https://refspecs.linuxbase.org/LSB_5.0.0/LSB-Core-generic/LSB-Core-generic/symversion.html#VERDEFEXTS
+#[derive(Debug)]
+pub struct Verdaux {
+    /// Offset to the version or dependency name string in the section header, in bytes.
+    pub vda_name: usize,
+    /// Offset to the next verdaux entry, in bytes.
+    pub vda_next: u32,
+}
 
-    /**************************
-     *  Version Requirements  *
-     **************************/
+/**************************
+ *  Version Requirements  *
+ **************************/
+
+/// Helper struct to iterate over [Version Needed][Verneed] and [Version Needed
+/// Auxiliary][Vernaux] entries.
+#[derive(Debug)]
+pub struct VerneedSection<'a> {
+    /// String table used to resolve version strings.
+    pub verstr: Strtab<'a>,
+    bytes: &'a [u8],
+    count: usize,
+    ctx: container::Ctx,
+}
 
-    /// Helper struct to iterate over [Version Needed][Verneed] and [Version Needed
-    /// Auxiliary][Vernaux] entries.
-    #[derive(Debug)]
-    pub struct VerneedSection<'a> {
-        /// String table used to resolve version strings.
-        pub verstr: Strtab<'a>,
+impl<'a> VerneedSection<'a> {
+    /// Try to parse the optional [`SHT_GNU_VERNEED`] section.
+    pub fn parse(
         bytes: &'a [u8],
-        count: usize,
+        shdrs: &'_ [SectionHeader],
```

Comment:
I'm a bit surprised you have to add the `'_` annotation; does default `&[SectionHeader]` lifetime elision not compile ?

Reply:
You are right, this should not be required.
I'll remove those.